src/realm/query.hpp

```diff
@@ -274,6 +274,10 @@ class Query final {
     Decimal128 maximum_decimal128(ColKey column_key, ObjKey* return_ndx = nullptr) const;
     Decimal128 minimum_decimal128(ColKey column_key, ObjKey* return_ndx = nullptr) const;
     Decimal128 average_decimal128(ColKey column_key, size_t* resultcount = nullptr) const;
+    Decimal128 sum_mixed(ColKey column_key) const;
```

Comment:
Could the return type not be Mixed? We could start out as int and then promote to double/Decimal128 as required.

Reply:
I chose to always use Decimal128 for sum and average of mixed because it was easier to implement and provides consistent results. It is also possible to dynamically choose the type based on the type of values we accumulate. If there is a problem with this approach, I am open to changing it. I don't have a concrete test case to prove this, but I suspect that with the dynamic type approach, the accumulation could produce different results on the same data depending on the ordering of the values.